tests/rptest/tests/compaction_e2e_test.py

```diff
@@ -31,11 +31,13 @@ def __init__(self, test_context):
               self).__init__(test_context=test_context,
                              extra_rp_conf=extra_rp_conf)
 
-    def topic_segments(self):
-        storage = self.redpanda.node_storage(self.redpanda.nodes[0])
-        topic_partitions = storage.partitions("kafka", self.topic)
-
-        return [len(p.segments) for p in topic_partitions]
+    def topic_segments(self) -> list[list[int]]:
+        partitions = []
+        for node in self.redpanda.nodes:
```

Comment:
nit: there's an API for getting the segments from the whole cluster by node. Just pointing it out; I'm not suggesting you change the current code..
```
self.redpanda.storage().segments_by_node()
```